### PR TITLE
GROWTH-147 iOS LTV

### PR DESCRIPTION
**Pull request by kwindau**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3478)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_ios/ltv_states/view.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{% set max_weeks = 32 %}
+{% set death_time = 160 %}
+{% set lookback = 28 %}
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.firefox_ios.ltv_states`
+AS
+SELECT
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/firefox_ios/ltv_states/view.sql:7]: I think this view is missing the actual state, right? i.e. what is the state of this user on this day

Comment (kwindau) [reply sql/moz-fx-data-shared-prod/firefox_ios/ltv_states/view.sql:7]: Yeah that's a good point, let me add that in as a new column

File: sql/moz-fx-data-shared-prod/firefox_ios/ltv_states/view.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  days_since_seen,
+  pattern,
+  active,
+  ad_clicks,
+  adjust_network,
+  first_reported_country,
+  first_reported_isp,
+  death_time,
+  max_weeks,
+  mozfun.ltv.get_state_ios_v2(
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/firefox_ios/ltv_states/view.sql:48]: WDYT about doing this the way we did with Android? i.e. explicitly naming states based on the state function, and accommodating multiple  https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/fenix/ltv_states/view.sql#L21-L22

**Review by fbertsch — APPROVED** [review]
r+wc

**Pull request merged**